### PR TITLE
Eio.Net: add getnameinfo

### PR DESCRIPTION
**Pull request by bikallem**

### CODE REVIEW

File: lib_eio_linux/eio_linux.ml [hunk 1]
```diff
@@ -909,6 +909,22 @@ module Low_level = struct
     Eio_unix.run_in_systhread @@ fun () ->
     Unix.getaddrinfo node service []
     |> List.filter_map to_eio_sockaddr_t    
+
+  let getnameinfo ~ni_dgram sockaddr =
+    let (addr,port) =
+      match sockaddr with
+      | `Unix _ -> raise (Invalid_argument "getnameinfo: Unix domain socket not supported")
```
Comment (talex5) [inline lib_eio_linux/eio_linux.ml:916]: OCaml's Unix module allows this:
```
# Unix.(getnameinfo (ADDR_UNIX "/tmp/sock") []);;
- : Unix.name_info = {Unix.ni_hostname = "debian"; ni_service = "/tmp/sock"}

```

File: lib_eio/net.mli [hunk 2]
```diff
@@ -105,6 +105,7 @@ class virtual t : object
   method virtual connect : sw:Switch.t -> Sockaddr.stream -> <stream_socket; Flow.close>
   method virtual datagram_socket : sw:Switch.t -> Sockaddr.datagram -> <datagram_socket; Flow.close>
   method virtual getaddrinfo : service:string -> string -> Sockaddr.t list
+  method virtual getnameinfo : ni_dgram:bool -> Sockaddr.t -> (string * string)
```
[the remark this answers is not in the document]
Comment (bikallem) [reply lib_eio/net.mli]: Removed.

File: lib_eio_linux/eio_linux.ml (hunk 1 above)
Comment (bikallem) [reply lib_eio_linux/eio_linux.ml]: Unix domain socket is now supported.

**Comment by bikallem** [comment]
@talex5 I have removed `ni_dgram` from the API. Since `luv` doesn't support getnameinfo on UNIX domain, we now use `Eio_unix.getnameinfo` on both uring and luv backends.

**Review by talex5 — APPROVED** [review]
Looks good - thanks!

**Pull request merged**